### DISCUSSION TITLE
Sort by distance

### DISCUSSION THREAD

**Discussion by j-kia** [directus/directus#15949] [Feature Requests]
I need to sort my results by distance based on a specified lat/lng provided in the query. I want to keep all the functionalities of the "items" endpoint but add a custom "sort". I know there is an hook for sort, but not sure if that would help.

**Comment by rijkvanzanten** [comment]
There's currently no way to calculate the distance on the fly. It would be great to add a `distance()` [function](https://docs.directus.io/reference/query.html#functions) which you could then use in `sort` or any other query parameter 🙂

**Reply by j-kia** [reply]
Thanks for your answer.
Is there any way to add a custom function? Or can we create a custom endpoint that uses SQL statement to read records ordered by distance? (I want, it still has the `ItemsService` functionalities)

**Reply by haniefss** [reply]
> There's currently no way to calculate the distance on the fly. It would be great to add a `distance()` [function](https://docs.directus.io/reference/query.html#functions) which you could then use in `sort` or any other query parameter 🙂

I think it'll be quite heavy if you were to do it on the fly. There are several formulas for measuring distance between two coordinates and all of them are a little complex. I'm currently implementing this myself (not for sorting though) by using a hook to calculate distance and store the results in a field, a `distance` field. I think it'll be easier for the system to sort a normal field.

I'm using this package https://www.npmjs.com/package/geographiclib-geodesic

**Reply by j-kia** [reply]
Thank for your help. In my case, one of the two coordinates that is required to measure the distance, can be changed constantly. As this is a mobile app and the user's location may change. Whenever the user's location changes, the `distance` field must be updated.

**Reply by rijkvanzanten** [reply]
> Whenever the user's location changes, the distance field must be updated.

And therefore when you have multiple users, you'd end up with multiple fields. While it's theoretically the most performant, it's not going to be scalable for this use case :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
